tree.go

```diff
@@ -937,15 +904,21 @@ func (n *LeafNode) ToHashedNode() *HashedNode {
 	return &HashedNode{commitment: comm[:]}
 }
 
-func (n *LeafNode) Insert(k []byte, value []byte, _ NodeResolverFn) error {
+func (n *LeafNode) Insert(key []byte, value []byte, _ NodeResolverFn) error {
+	if len(key) != StemSize+1 {
+		return fmt.Errorf("invalid key size: %d", len(key))
+	}
+	if !bytes.Equal(key[:StemSize], n.stem) {
+		return fmt.Errorf("stems doesn't match: %x != %x", key[:StemSize], n.stem)
+	}
 	values := make([][]byte, NodeWidth)
-	values[k[31]] = value
-	return n.insertMultiple(k[:31], values)
+	values[key[StemSize]] = value
+	return n.insertMultiple(key, values)
```

Comment:
Do you mean 

```suggestion
	return n.insertMultiple(key[:StemSize], values)
```

? You're passing 32 bytes instead of 31, but it is compared to a stem in `insertMultiple`.

Reply:
`insertMultiple(...)` first parameter is named `key` so passing a key (32 bytes) it should be fine, since the method should handle that semantics. `insertMultiple(...)` calls `equalPaths(...)` which already does the `[:StemSize]` accordingly.

Reply:
OK, I see your comments below. If you want to change from `key` to `stem`, then that suggested change is needed. 
Just replied that isn't the case without changing the API.

I can change from `key` to `stem` for sure.